src/sdk.spec.js

```diff
@@ -1,86 +1,43 @@
-import {act} from '@testing-library/react';
+import {act, waitForElementToBeRemoved, within} from '@testing-library/react';
 
-import {OpenForm} from './sdk.js';
+import {BASE_URL, getForm, mockFormGet} from 'api-mocks';
+import mswServer from 'api-mocks/msw-server';
+import {mockFormioTranslations, mockLanguageInfoGet} from 'components/LanguageSelection/mocks';
 
-// TODO: look into msw for API mocking instead
-// see https://testing-library.com/docs/react-testing-library/example-intro
-const apiModule = require('api');
-jest.mock('api');
+import {OpenForm} from './sdk.js';
 
 // scrollIntoView is not not supported in Jest
 let scrollIntoViewMock = jest.fn();
 window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;
 
-const FORM = {
-  uuid: '81a22589-abce-4147-a2a3-62e9a56685aa',
-  url: 'http://localhost:8000/api/v2/forms/81a22589-abce-4147-a2a3-62e9a56685aa',
-  name: 'MOCKED',
-  active: true,
-  loginRequired: false,
-  loginOptions: [],
-  product: '',
-  slug: 'jest',
-  maintenanceMode: false,
-  showProgressIndicator: true,
-  submissionAllowed: 'yes',
-  literals: {
-    beginText: {resolved: 'begin', value: ''},
-    changeText: {resolved: 'change', value: ''},
-    confirmText: {resolved: 'confirm', value: ''},
-    previousText: {resolved: 'previous', value: ''},
-  },
-  steps: [],
-  explanationTemplate: '',
-  requiredFieldsWithAsterisk: true,
-  autoLoginAuthenticationBackend: '',
-  translationEnabled: true,
-};
-
-const LANGUAGE_INFO = {
-  languages: [
-    {code: 'nl', name: 'Nederlands'},
-    {code: 'en', name: 'English'},
-  ],
-  current: 'nl',
-};
-
-const mockAPICalls = url => {
-  switch (url) {
-    case 'http://localhost:8000/api/v2/forms/81a22589-abce-4147-a2a3-62e9a56685aa': {
-      return FORM;
-    }
-    case 'http://localhost:8000/api/v2/i18n/formio/nl': {
-      return {};
-    }
-    case 'http://localhost:8000/api/v2/i18n/info': {
-      return LANGUAGE_INFO;
-    }
-    default:
-      throw new Error(`Unknown url: ${url}`);
-  }
-};
+const LANGUAGES = [
+  {code: 'nl', name: 'Nederlands'},
+  {code: 'en', name: 'English'},
+];
+
+const apiMocks = [
+  mockFormGet(getForm({translationEnabled: true})),
+  mockLanguageInfoGet(LANGUAGES),
+  mockFormioTranslations,
+];
 
 describe('OpenForm', () => {
   it('should accept a DOM node as languageSelectorTarget', async () => {
+    mswServer.use(...apiMocks);
     const formRoot = document.createElement('div');
     const target = document.createElement('div');
-
-    apiModule.get.mockImplementation(mockAPICalls);
-
     const form = new OpenForm(formRoot, {
-      baseUrl: 'http://localhost:8000/api/v2/',
+      baseUrl: BASE_URL,
       basePath: '',
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       languageSelectorTarget: target,
       lang: 'nl',
     });
 
-    try {
-      await act(async () => await form.init());
-    } catch (e) {
-      throw e; // should not error
-    }
+    await act(async () => await form.init());
```

Comment:
tried out throwing exceptions in various places and they all seem to bubble up and fail/error the test itself, so I don't think this original `try` wrapping was necessary, unless I missed something?

Reply:
Probably copy pasted.